### PR TITLE
Angular 6 update

### PR DESCRIPTION
**Pull request by sjimenez77**
- Dependencies updated.
- Scripts updated and fixed in package.json. 
- Deprecated date and time picker updated and localized. 
- Some Angular Material 6 fixes. 
- PWA configuration updated.

### CODE REVIEW

File: angular/src/app/book-table/book-table.component.html [hunk 1]
```diff
@@ -81,60 +78,53 @@ <h3>{{'bookTable.title' | translate}}</h3>
...
               </div>
 
               <h4>{{'formFields.guests' | translate}}</h4>
-              <td-chips
-                class="pad-bottom-sm"
-                formControlName="invitedGuests"
-                name="invitedGuests"
-                (add)="validateEmail()"
-                [placeholder]="'formFields.guestsPH' | translate">
+              <td-chips class="pad-bottom-sm" formControlName="invitedGuests" name="invitedGuests" (add)="validateEmail()" [placeholder]="'formFields.guestsPH' | translate">
```
Comment (dumbNickname) [inline angular/src/app/book-table/book-table.component.html]: To be honest I would try to keep formatting consistent (i preferred the one before - if there are many attributes, then it is much more clear to list those line by line, later also easier for git to find differences when one attribute is added / changed)

Comment (sjimenez77) [reply angular/src/app/book-table/book-table.component.html]: Ok, Agree. Well, to be honest this line has been modified by prettier, but I am committing it with your suggestion.

File: angular/src/app/book-table/book-table.component.ts [hunk 2]
```diff
@@ -34,78 +32,111 @@ export class BookTableComponent implements OnInit {
...
+      ),
+      name: new FormControl(
+        this.reservationInfo.booking.name,
         Validators.required,
-        Validators.pattern(/^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i),
+      ),
+      email: new FormControl(this.reservationInfo.booking.email, [
+        Validators.required,
+        Validators.pattern(
+          /^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i,
```
Comment (dumbNickname) [inline angular/src/app/book-table/book-table.component.ts]: Is this the same email pattern repeated twice? I know that not a part of your task, but will note it down to fix later

File: angular/src/app/book-table/book-table.module.ts [hunk 3]
```diff
@@ -20,28 +19,18 @@ import { TranslateModule } from '@ngx-translate/core';
...
     HttpClientModule,
     TranslateModule,
     CoreModule,
   ],
-  providers: [
-    BookTableService,
-    WindowService,
-    SnackBarService,
-  ],
+  providers: [BookTableService, WindowService, SnackBarService],
```
Comment (dumbNickname) [inline angular/src/app/book-table/book-table.module.ts]: I would have that small remark with formatting again here.

File: angular/src/app/book-table/book-table.component.ts (hunk 2 above)
Comment (sjimenez77) [reply angular/src/app/book-table/book-table.component.ts]: Ok, perfect 👍

**Comment by sjimenez77** [comment]
imagen

**Comment by cbeldacap** [comment]
Already successfully tested in Production Line 👍

**Pull request merged**